src/main/java/bio/terra/service/snapshotbuilder/query/Query.java

```diff
@@ -33,21 +35,26 @@ public record Query(
   }
 
   public Query(List<FieldVariable> select, List<TableVariable> tables) {
-    this(select, tables, null, null, null, null);
+    this(select, tables, null, null, null, null, null);
   }
 
   public Query(
       List<FieldVariable> select, List<TableVariable> tables, List<FieldVariable> groupBy) {
-    this(select, tables, null, groupBy, null, null);
+    this(select, tables, null, groupBy, null, null, null);
   }
 
   public Query(List<FieldVariable> select, List<TableVariable> tables, FilterVariable where) {
-    this(select, tables, where, null, null, null);
+    this(select, tables, where, null, null, null, null);
   }
 
+  // where limit is added, platform must be added as well
```

Comment:
Instead of a comment here, can you add an assertion check to the `Query` constructor above? Something like
```
    if (limit != null && platform == null) {
      throw new IllegalArgumentException("...");
    }
```